### PR TITLE
Migrate plugin manifest references from yaml to JSON

### PR DESCRIPTION
**Pull request by hanzei**
This PR migrates all plugin manifest related docs to use JSON for the manifest. 

Ref https://github.com/mattermost/mattermost-server/issues/10173

### CODE REVIEW

File: site/content/extend/plugins/webapp/hello-world.md [hunk 1]
```diff
@@ -117,15 +117,14 @@ Now, we'll need to define the required manifest describing your plugin's entry p
...
-    "description": "",
+    "name": "Hello World",
     "webapp": {
         "bundle_path": "main.js"
     }
 }
 ```
 
-This manifest gives the server the location of our web app  within our bundle. (Note that you may alternatively use `plugin.yaml`, as shown in [../../server/hello-world/](../../server/hello-world/).)
+This manifest gives the server the location of our web app  within our bundle. (Note that you may alternatively use `plugin.yaml`) Consult the [manifest reference]({{< ref "manifest-reference.md" >}}) for more details, including how to define a minimum required server version for your plugin.
```
Comment (jasonblais) [inline site/content/extend/plugins/webapp/hello-world.md]: Is the first sentence accurate - it gives the location of `the web app` within the plugin bundle? Or is the webapp components of the plugin?

Comment (jasonblais) [reply site/content/extend/plugins/webapp/hello-world.md]: If it is, propose tweaking this section to

```
This manifest gives the server the location of our web app within our plugin bundle. Consult the [manifest reference]({{< ref "manifest-reference.md" >}}) for more details, including how to define a minimum required server version for your plugin.

Note that you may also use `plugin.yaml` to define the manifest.
```

File: site/content/extend/plugins/server/hello-world.md [hunk 2]
```diff
@@ -57,21 +57,24 @@ GOOS=linux GOARCH=amd64 go build -o plugin.exe plugin.go
...
+    "id": "com.mattermost.server-hello-world",
+    "name": "Hello World",
+    "server": {
+        "executable": "plugin.exe"
+    }
+}
 ```
 
-This manifest gives the server the location of our executable within our bundle. (Note that you may alternatively use `plugin.json`, as shown in [webapp/hello-world/](../../webapp/hello-world/).) Consult the [manifest reference](/extend/plugins/manifest-reference/) for more details, including how to define a cross-platform bundle by defining multiple executables.
+This manifest gives the server the location of our executable within our bundle. (Note that you may alternatively use `plugin.yaml`) Consult the [manifest reference]({{< ref "manifest-reference.md" >}}) for more details, including how to define a cross-platform bundle by defining multiple executables or how to define a minimum required server version for your plugin.
```
Comment (jasonblais) [inline site/content/extend/plugins/server/hello-world.md]: Propose

```
This manifest gives the server the location of our executable within our plugin bundle. Consult the [manifest reference]({{< ref "manifest-reference.md" >}}) for more details, including how to define a cross-platform bundle by defining multiple executables, and how to define a minimum required server version for your plugin.

Note that you may also use `plugin.yaml` to define the manifest.
```

File: site/content/extend/plugins/webapp/hello-world.md (hunk 1 above)
Comment (hanzei) [reply site/content/extend/plugins/webapp/hello-world.md]: Fixed

File: site/content/extend/plugins/server/hello-world.md (hunk 2 above)
Comment (hanzei) [reply site/content/extend/plugins/server/hello-world.md]: Fixed

**Pull request merged**